### ISSUE TITLE
L1T Private Integration Branch based on CMSSW_8_0_8

### ISSUE THREAD

**Issue by mulhearn** [cms-l1t-offline/cmssw#318]
The 808 integration branch has been adjusted to better line up with what will be in CMSSW, by rebasing to CMSSW 808 plus these pending 80x PRs:
- main v6 pr 80x cms-sw#14362
- ntuples v6 80x: cms-sw#14493
- HCAL TP PR 80x: cms-sw#14419
- main v7 pr 80x: cms-sw#14570

The branch is located here:
   https://github.com/cms-l1t-offline/cmssw/tree/l1t-integration-CMSSW_8_0_8

Bitwise agreement in integration tests demonstrated between l1t-integration-v58.1-CMSSW_8_0_8 and l1t-integration-v58.1 (under 807).   And likewise for v54.0.  (Note that l1t-tsg-v7-cand = l1t-integration-v54.0)

Tags:
- l1t-integration-v58.1-CMSSW_8_0_8 is equivalent to l1t-integration-v58.1 (under 807)
- l1t-integration-v54.0-CMSSW_8_0_8 is equivalent to l1t-integration-v54.0 (under 807)

Recipe:

```
cmsrel CMSSW_8_0_8
cd CMSSW_8_0_8/src
cmsenv
git cms-init
git remote add cms-l1t-offline git@github.com:cms-l1t-offline/cmssw.git
git fetch cms-l1t-offline
git cms-merge-topic --unsafe cms-l1t-offline:l1t-integration-v61.1-CMSSW_8_0_8
scram b -j 8
```

The change log is here:

https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideL1TOfflineDev#Development_Integration

**Comment by mulhearn** [comment]
I'm detecting bitwise differences for muons with our proto PR for l1t-tsg-v7-cand into 81x (80x is fine).

I guessed it is due to this:

```
mulhearn@cmsdev01>git log l1t-tsg-v6-cand..HEAD -- L1Trigger/CSCTriggerPrimitives | more
commit 53cf53ad6a34c9c4ebee00f5ef022ce29b366489
Author: Sven Dildick <sven.dildick@cern.ch>
Date:   Fri May 13 09:09:57 2016 -0500

    use sys.stderr.write instead of print

commit f17a955a3e45d9c3a4cfd133f4f0a7621c388b88
Author: Sven Dildick <sven.dildick@cern.ch>
Date:   Mon May 9 07:13:47 2016 -0500

    temporary hack to run on data

commit 65365e9b60f6335c7e8f614b2e9f5a002b550db1
Author: Sven Dildick <sven.dildick@cern.ch>
Date:   Sat May 7 08:09:35 2016 +0200

    temporary hack to ensure that all data MPC LCTs are read out

commit fb0b2f98f8ac0dbfdbc416c434482d3ba01d10f5
Author: Sven Dildick <sven.dildick@cern.ch>
Date:   Sat May 7 00:22:20 2016 +0200

    options to drop invalid and low quality stubs

commit f1d5d4c4af8570a2bc36039171023db33c2de152
Author: Sven Dildick <sven.dildick@cern.ch>
Date:   Thu May 5 23:25:37 2016 +0200

    bugfixes to CSC TP Run2 settings

```

and indeed when those are reverted, we restore bitwise agreement.

**Comment by blwiner** [comment]
@Mulhearn did PR https://github.com/cms-l1t-offline/cmssw/pull/299 ever go
in 8_0_7?  I see it still sitting there.  The MinBias part was cherry
picked, but we still have the word definition change.  This was desired by
@tmatsush  for firmware and operation at P5

 -Brian

**Comment by tmatsush** [comment]
@mulhearn @blwiner 
as we have not heard that the word definition change is ready in offline, the definition has not changed in firmware yet. We can switch at anytime when it becomes ready.

**Comment by fwyzard** [comment]
if you are going to make changes in the firmware that affect how the data in unpacked and/or emulated, is there a way of "versioning" the changes, so that both old and new data can be unpacked/emulated ?

**Comment by mulhearn** [comment]
Yes, we have such mechanism, but in this case, I think we can just make the change and no one will notice.  This only effects intermediate bits that no one is looking at.

**Comment by mulhearn** [comment]
Looks like Rekovic missed my ping.  We'll just put this in immediately and it will go in with l1t-tsg-v7-cand PR today.

**Comment by thomreis** [comment]
@mulhearn could you post a working cmsDriver configuration please? Mine from 8_0_7 gives a productNotFound exception.

> %MSG
> ----- Begin Fatal Exception 17-May-2016 14:15:38 CEST-----------------------
> An exception of category 'ProductNotFound' occurred while
>    [0] Processing run: 260627 lumi: 107 event: 150825591
>    [1] Running path 'l1ntupleaod'
>    [2] Calling event method for module L1TauRecoTreeProducer/'l1TauRecoTree'
> Exception Message:
> Principal::getByToken: Found zero products matching all criteria
> Looking for type: reco::PFTauDiscriminator
> Looking for module label: hpsPFTauDiscriminationByMVA6VLooseElectronRejection
> Looking for productInstanceName:

**Comment by mulhearn** [comment]
@blwiner, that commit is now in integration branch at v53.2 in 807.  I'll forward port it here as well.  It has no effect on our integration tests, as I expected, since no one is looking at those intermediate bits.

**Comment by mulhearn** [comment]
This branch is now fully up to date with Vladimir's work on 807 and bitwise equivalent performance in integration tests demonstrated.  Its ready for widespread use and we can deprecate the 807 integration branch.

**Comment by mulhearn** [comment]
Added some new tags:
- v59.1 - remove now unneeded condition overrides (now default) for geometry and HF1x1s, still used for 2015 re-emulation.
- v60.0 - add sum overflow check for et and ht
- v61.0 - switch to tau isolation option 21
- v61.1 - add new L1REPACK:FullMC
